fgpyo/sam/__init__.py

```diff
@@ -111,6 +111,25 @@
     >>> [e.operator.is_indel for e in cigar.elements]
     [False, True, False, True, False]
 
+Any particular tuple can be accessed directly with its index (and works with negative indexes and slices):
+
+    >>> cigar = Cigar.from_cigarstring("50M2D5M2I10S")
+    >>> cigar[0].length
+    50
+    >>> cigar[1].operator
+    <CigarOp.D: (2, 'D', False, True)>
+    >>> cigar[-1].operator
+    <CigarOp.S: (4, 'S', True, False)>
+    >>> tuple(x.operator.character for x in cigar[1:3])
+    ('D','M')
+    >>> tuple(x.operator.character for x in cigar[-2:])
+    ('I', 'S')
+
+
+The Cigar has __len__ defined to be the length of the elements:
+    >>> cigar = Cigar.from_cigarstring("50M2D5M2I10S")
+    >>> len(cigar) == len(cigar.elements)
+    True
```

Comment:
I actually would prefer we don't define __len__.  I think it's likely to lead to confusion since we have so many different lengths in question:
- `len(cigar.elements)`
- `sum(c.length for c in cigar.elements)`
- `cigar.length_on_target()`
- `cigar.length_on_query()`